R/Plot_synergy.R

```diff
@@ -1,328 +1,332 @@
+# lists whose first two elements are range endpoints, and whose remaining
+# elements (if any) are ignored.
+.ApplyRanges <- function(data, row.range, col.range) {
+  # Default value is 'remove the element just after the end of the list',
+  # for each dimension of the array (a no-op that permits cleaner syntax).
+  return(data[.RangeWithDefault(row.range, default=-(nrow(data)+1)),
+              .RangeWithDefault(col.range, default=-(ncol(data)+1))])
+}
+
+# If 'range' is NULL, return 'default', otherwise return a slice.
+# The endpoints of the slice are determined by assuming 'range' is a list, and
+# picking its first two elements.  Any additional elements are ignored.
+.RangeWithDefault <- function(range, default) {
+  return(if(is.null(range)) default else range[1]:range[2])
+}
+
+
+# TODO/REVIEW: The original logic checked into GitHub at the time this function
+# was written used flipped argument order for data.coord and krig.coord.
+# Furthermore, that logic appeared to transpose its output.  Please see the
+# corresponding unit test in test_that("extended_scores_works", {...}).  The
+# original logic swapped the locations of corner values '2' and '3'.  Looking
+# at the dose/response matrices, it seems that this transposition may have been
+# a bug.  With the code below, the contour and wireframe plots are similarly
+# transposed from their current appearance in the documentation, but seem to
+# gibe better with the documented dose/response matrices.
+
+# Perform statistical interpolation to smooth plotted values.
+.ExtendedScores <- function(scores.dose, len) {
+  # len: how many values need to be predicted between two adjacent elements
+  #      of scores.dose
+  nr <- nrow(scores.dose)
+  nc <- ncol(scores.dose)
+  # Note, not all kriging implementations are suitable for all purposes.  In
+  # particular, SpatialExtremes offers an implementation that works with small
+  # data sets.  Earlier versions of this implementation that used other kriging
+  # packages only worked well with large ones.
+  return(SpatialExtremes::kriging(
+    data = c(scores.dose),
+    data.coord = cbind(rep(1:nr, nc), rep(1:nc, each = nr)),
+    krig.coord = cbind(seq(1, nc, length = (nc - 1)*(len + 2) - (nc - 2)),
+                       seq(1, nr, length = (nr - 1)*(len + 2) - (nr - 2))),
+    cov.mod = "whitmat",
+    grid = TRUE,
+    sill = 1,
+    range = 10,
+    smooth = 0.8
+  )$krig.est)
+}
+
+.Plot2D <- function(plot.title, mat.tmp, len, row.conc, col.conc,
+                    drug.row, drug.col, start.point, end.point) {
+  # Note a different layout() here than in .PlotAll()
+  layout(matrix(c(1, 2), nrow = 2L, ncol = 1L), heights = c(0.1, 1))
+  .Plot2DTitle(plot.title, start.point, end.point)
+  .Plot2DContourMap(
+    mat.tmp, row.conc, col.conc, drug.row, drug.col, start.point, end.point)
+  return(recordPlot())
+}
+
+.Plot2DTitle <- function (plot.title, start.point, end.point) {
```

Comment:
Since this function plots title and legend, it might be good to rename it as ".Plot2DTitleAndLegend".

Reply:
Done.